File: java/angular_spring_boot/backend/src/main/java/com/example/demo/controller/HelloController.java
package com.example.demo.controller;

import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@CrossOrigin(origins = "*")
public class HelloController {

    @GetMapping("/api/hello")
    public String hello() {
        return "Hello from Spring Boot!";
    }
}

File: kotlin/spring_boot/src/main/kotlin/com/example/demo/controller/HelloController.kt
package com.example.demo.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController {

    // http://localhost:8080/hello?name=test
    @GetMapping("/hello")
    fun sayHello(@RequestParam name: String): String {
        return "Hello, $name!"
    }
}